system/Database/OCI8/Forge.php

```diff
@@ -284,7 +284,7 @@ protected function _processForeignKeys(string $table): string
         ];
 
         foreach ($this->foreignKeys as $fkey) {
-            $nameIndex            = $table . '_' . implode('_', $fkey['field']) . '_fk';
+            $nameIndex            = $table . '_' . implode('_', $fkey['field']) . '_foreign';
```

Comment:
`_foreign` is too long for OCI8.
Oracle Database 12.1 can have only 30 byte object name.

Reply:
This change also changed the behavior of `createTable()`. It is also a breaking change.

We need to discuss.

Reply:
How about:

```php
            $nameIndex = 'fk_' . $table . '_' . implode('_', $fkey['field']);

            if (version_compare($this->db->getVersion(), "18.0.0", "<") && strlen($nameIndex) > 30) {
                if (strlen($nameIndex) > 30) {
                    $nameIndex = substr($nameIndex, 0, 28) . rand(10, 99);
                }
            }

            $nameIndexFilled      = $this->db->escapeIdentifiers($nameIndex);
```

We change them all to fk_

Reply:
Can't do a prefix.

Reply:
Why can't do prefix?